cmd/watch-termination/main.go

```diff
@@ -126,12 +127,31 @@ func run() int {
 			// keep going
 		}
 
-		defer func() {
+		var deleteLockOnce sync.Once
+
+		if *gracefulTerminatioPeriod > 2*time.Second {
+			go func() {
+				<-time.After(*gracefulTerminatioPeriod - 2*time.Second)
```

Comment:
how do you know when this `goroutine` will be scheduled?
it can be scheduled after 100s then this code would wait 100s *(`gracefulTerminatioPeriod - 2*time.Second`)

Reply:
fixed